sources/file-source/yamlconfigurationsource_test.go

```diff
@@ -87,44 +138,59 @@ testfilekey3:
 	check(err)
 	f2, err := os.Create(file2)
 	check(err)
+	f3, err := os.Create(file3)
+	check(err)
 	defer f1.Close()
 	defer f2.Close()
+	defer f3.Close()
 	defer os.Remove(file1)
 	defer os.Remove(file2)
+	defer os.Remove(file3)
 
 	_, err = io.WriteString(f1, string(f1content))
 	check(err)
 	_, err = io.WriteString(f2, f2content)
 	check(err)
+	_, err = io.WriteString(f3, string(file))
+	check(err)
 
 	fSource := NewYamlConfigurationSource()
 
+	var fileData interface{}
+
+	fSource.AddFileSource(&fileData, file3, 0)
```

Comment:
use real hystrix struct

Reply:
I mean you can write your own simple struct  and use this struct to test ,but not map